File: Programmers/Level2/[3차]방금그곡/[3차]방금그곡.py
def solution(m, musicinfos):
    answer = []
    m = m.replace('C#', 'c').replace('D#', 'd').replace('F#', 'f').replace('G#', 'g').replace('A#', 'a')
    
    for musicinfo in musicinfos:
        start, end, title, code = musicinfo.split(',')
        
        start_hour, start_minute = map(int, start.split(':'))
        end_hour, end_minute = map(int, end.split(':'))
        duration = (end_hour - start_hour) * 60 + end_minute - start_minute
        
        code = code.replace('C#', 'c').replace('D#', 'd').replace('F#', 'f').replace('G#', 'g').replace('A#', 'a')
        code *= (duration // len(code)) + 1
        code = code[:duration]
        
        if m in code:
            answer.append([duration, start_hour * 60 + start_minute, title])
    
    if answer:
        return sorted(answer, key=lambda x:(-x[0], x[1]))[0][-1]
    return "(None)"

File: Programmers/Level2/[3차]방금그곡/[3차]방금그곡.js
function solution(m, musicinfos) {
  const answer = [];
  m = m.replace(/(\D)#/g, (_, c) => c.toLowerCase());

  for (let musicinfo of musicinfos) {
    let [start, end, title, code] = musicinfo.split(",");

    let [start_hour, start_minute] = start.split(":");
    let [end_hour, end_minute] = end.split(":");
    const duration = (end_hour - start_hour) * 60 + +end_minute - start_minute;

    code = code.replace(/(\D)#/g, (_, c) => c.toLowerCase());
    code = code.repeat(Math.floor(duration / code.length) + 1);
    code = code.slice(0, duration);

    if (code.includes(m))
      answer.push([duration, start_hour * 60 + +start_minute, title]);
  }

  if (answer.length) {
    return answer.sort((a, b) => {
      if (a[0] > b[0]) return -1;
      if (a[0] < b[0]) return 1;
      if (a[1] > b[1]) return 1;
      if (a[1] < b[1]) return -1;
    })[0][2];
  }
  return "(None)";
}
